storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/serialization/VoteTrackerSerializer.java

```diff
@@ -28,7 +31,14 @@ public VoteTracker deserialize(final byte[] data) {
           final Bytes32 currentRoot = Bytes32.wrap(reader.readFixedBytes(Bytes32.SIZE));
           final Bytes32 nextRoot = Bytes32.wrap(reader.readFixedBytes(Bytes32.SIZE));
           final UInt64 nextEpoch = UInt64.fromLongBits(reader.readUInt64());
-          return new VoteTracker(currentRoot, nextRoot, nextEpoch);
+          if (reader.isComplete()) {
+            // Old version
+            return new VoteTrackerV1(currentRoot, nextRoot, nextEpoch);
+          } else {
+            // Version with equivocation
+            return new VoteTrackerV2(
+                currentRoot, nextRoot, nextEpoch, reader.readBoolean(), reader.readBoolean());
+          }
```

Comment:
We deliberately read values on a separate line to the constructor so the reads don't get reordered if the constructor parameter order is refactored.

And this can just build a single version like:
```suggestion
          final boolean currentEquivocating;
          final boolean nextEquivocating;
          if (reader.isComplete()) {
            currentEquivocating = false;
            nextEquivocating = false;
          } else {
            currentEquivocating = reader.readBoolean();
            nextEquivocating = reader.readBoolean();
          }
          return new VoteTracker(currentRoot, nextRoot, nextEpoch, currentEquivocating, nextEquivocating);
```

Actually we probably want to know if equivocating validators has been disabled and read the booleans but then set them to false anyway.